replication-api-impl/src/test/java/com/connexta/replication/api/impl/data/FilterIndexManagerImplTest.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.api.impl.data;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.connexta.ion.replication.api.NotFoundException;
+import com.connexta.ion.replication.api.ReplicationPersistenceException;
+import com.connexta.replication.api.data.FilterIndex;
+import com.connexta.replication.api.impl.persistence.spring.FilterIndexRepository;
+import com.connexta.replication.api.persistence.FilterIndexManager;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FilterIndexManagerImplTest {
+
+  private FilterIndexManager sites;
+
+  @Mock FilterIndexRepository repository;
+
+  @Before
+  public void setup() {
+    sites = new FilterIndexManagerImpl(repository);
+  }
+
+  @Test(expected = NotFoundException.class)
+  public void testGetBySiteNotFound() {
+    final String id = "id";
+    when(repository.findById(id)).thenReturn(Optional.empty());
+    sites.get(id);
+  }
+
+  @Test(expected = NotFoundException.class)
+  public void testGetBySiteIdNotFound() {
```

Comment:
✏️ GetByFilter?

Reply:
❗️ Test when one is found, when a pojo is invalid

Reply:
Aren't I really testing the FilterIndexImpl readFrom ctor at that point in this test?

Reply:
If you want to mock the readFrom() to achieve it then fine. At the end of the day, you would be testing the contract of this constructor. The fact that it is currently implemented by calling readFrom() is beside the fact as we need to make sure it does what it is supposed to.

Reply:
Should we be testing the ctor of that class from within this manager test class

Reply:
Here you are not testing the ctor, you are just making sure that if the ctor/deserialization throws a persistence exception, it will come out from the find